components/main_menu/main_menu.tsx

```diff
@@ -459,7 +473,42 @@ export class MainMenu extends React.PureComponent<Props> {
                         <Menu.ItemLink
                             id='createTeam'
                             to='/create_team'
+                            className={createTeamRestricted ? 'MenuItem__with-icon-tooltip' : ''}
+                            disabled={this.props.isCloudFreeEnabled && this.props.teamsLimitReached}
```

Comment:
Same here, all cloud checks should be against the cloud license not just the feature flag